new-site/docs-site/content/0.19.0/guide/typesense-dynamodb.md

```diff
@@ -0,0 +1,297 @@
+# DynamoDB Integration with Typesense
+
+Hey there! This post give you a guide to integrate Typesense custer with AWS DynamoDB by setting up a trigger with AWS Lambda.
+
+![Typesense DynamoDB Integration Chart](~@guide-images/typesense-dynamodb/dynamodb-typesense-integration-1.png)
+
+## Create AWS DynamboDB with streams enabled
+
+Create a DynamoDB with your choice of name and partition key ("id" is recommended). Now, after creating a DynamoDB database you should enable streams in the "Overview" section.
+
+![DynamoDB streams](~@guide-images/typesense-dynamodb/dynamodb-typesense-integration-2.png)
+
+If you are doing this in AWS CLI,
+
+```bash
+aws dynamodb create-table \
+    --table-name typensense \
+    --attribute-definitions AttributeName=id,AttributeType=N \
+    --key-schema AttributeName=id,KeyType=HASH  \
+    --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5 \
+    --stream-specification StreamEnabled=true,StreamViewType=NEW_AND_OLD_IMAGES
+```
+## Setting up Triggers for DyanmoDB
+
+Create a trigger for DynamoDB database and this will lead you to AWS Lambada function's page from there you can select an existing funciton or a new function. In this tutorial the function name is ```typesense-indexing```
+
+![DynamoDB Triggers](~@guide-images/typesense-dynamodb/dynamodb-typesense-integration-3.png)
+
+Then now you need to create a "Lambda Execution Role" i.e give permission for your function. Now, head over to IAM Roles section and create a new role with three main permissions:
+
+* AmazonDynamoDBFullAccess
+* AmazonDynamoDBFullAccesswithDataPipeline
+* AWSLambdaBasicExecutionRole
+
+![IAM Execution Role](~@guide-images/typesense-dynamodb/dynamodb-typesense-integration-4.png)
+
+Now, attach this execution role to the trigger Lambda funtion of DynamoDB.
```

Comment:
Let's add a warning here saying: These IAM permissions are just examples for the purposes of this guide. In a production setting, please consult the IAM documentation to only grant the minimal permissions needed for your particular use case.